### ISSUE TITLE
Cashier UserModel not returned by Auth::getUser?

### ISSUE THREAD

**Issue by therealkevinard** [OFFLINE-GmbH/oc-cashier-plugin#5]
Maybe I missed some tiny something in the docs, but my `Auth::getUser()` isn't returning the Cashier user model (with the obvious side-effects re getting invoices and things like that). 

Up to now, I've been okay with (basically proxying) and converting the usual user by `CashierUser::whereId(Auth::getUser()->id)`, but it's leading to a lot of extra code. 

Before forking with a new behavior, i thought I'd ask here.

**Comment by therealkevinard** [comment]
The code all seemed solid, but nothing I did would return the Cashier User. I have several custom plugins for this site anyway, so I replicated your `register` method to my 'main' plugin's `register`. 

Works like a charm now. 

This closes my personal issue.

**Comment by AugmentBLU** [comment]
Same issue here, `Auth` is not being bound or something. I even tried using the register function in my main plugin and still nothing. 

Can we debug some how like this?

```
public function register()
    {
        App::singleton('user.auth', function () {dd('f');
            return \Isa\Cashier\Classes\AuthManager::instance();
        });
        dd(\Auth::getUser());
    }
```

**Comment by damsfx** [comment]
Same issue here with `Auth` service.  
If I follow your [documentation's exemple](https://github.com/OFFLINE-GmbH/oc-cashier-plugin/blob/master/README.md#handle-form-submission)  :
```php
public function boot()
{
    parent::boot();

    Event::listen('offline.cashier::stripeElementForm.submit', function ($post) {
        $token = $post['token']['id'] ?? null;
        if ( ! $token) {
            throw new \RuntimeException('Stripe token is missing!');
        }

        $user = \Auth::getUser();
        $user->newSubscription('main', 'users-selected-plan-id')->create($token);

        return [
            'redirect' => \Url::to('thank-you')
        ];
    });
}
```
User is still an iteration of `RainLab\User\Models\User` !!

This is the first of your plugins for which integration is going quite smoothly.

**Comment by tobias-kuendig** [comment]
This is a known problem. The order in which October loads the plugins is unclear, that's why the cashier plugin might override your container bindings again, as it may be loaded after your plugin.

This issue was raised again a few days ago and it looks like this time it is getting addressed:
https://github.com/octobercms/october/issues/4826

**Comment by tobias-kuendig** [comment]
Looking at this issue again, if you move the contents of the `register` method to the `boot` method, does that work for you?

https://github.com/OFFLINE-GmbH/oc-cashier-plugin/blob/master/Plugin.php#L28-L30

**Comment by damsfx** [comment]
Yes ... it work like a charm.

`\Auth::getUser();` return an `\OFFLINE\Cashier\Models\User` in event manager and in `[invoicesList]` component.

**Comment by tobias-kuendig** [comment]
Nice! Thanks for testing, I'll release an update with this fix.

**Comment by tobias-kuendig** [comment]
The fix has been released as version 1.0.7, thank you for the feedback @damsfx!
